pipeline/tests/dune

```diff
@@ -0,0 +1,11 @@
+(test
+ (name test)
+ (libraries alcotest alcotest-lwt current current.fs current_docker
+   current_git current_github current_ocluster capnp-rpc-unix current_web
+   dockerfile fmt.tty logs logs.fmt prometheus rresult uri yojson str)
+ (preprocess
+  (pps ppx_deriving_yojson)))
+
+; copy everything but *.pp.ml
+
+(copy_files ../lib/**[!.]??.ml)
```

Comment:
Why do we need to copy these files? Can we just have public modules instead?

Reply:
Yeah this copy is the "hacky" part. The proper way would be to expose all the modules and functions that we want to test in the `pipeline.mli` library, which is a bit tedious:

```ocaml
module Api : sig
  val authenticate_token : ...
  val authorize_token : ...
end
...
```

But the main issue is that we can't mock the postgresql module if we do that, so it becomes a lot harder to test the side effects part. By copying, we get "free" access to the internal functions for testing **and** we gain the ability to remove/simulate the side effectful libraries :)

(The copy is fairly benign as dune copies the files in the hidden `_build/` folder, not the `tests/` directory that you interact with)